### ISSUE TITLE
[a11y]: UI Shell [Header & Header with Navigation] - current links bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15270]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.1

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
Current state of the link is not announced by screen reader

Link is visually looking link selected but the screen reader is not announcing the same. Note: In Header with navigation, action and side nav, same issue is observed for the buttons in the side navigation bar.

### Recommended Fix
Screen reader should also announce about the selection of the elements.

People who uses screen reader will not be aware of the state of the element.

### WCAG 2.1 Violation
4.1.2 Name, Role, Value
